File: Practice/atcoder/ABC/031/src/c.py
# N<=50なので累積和を使って全探索していく感じ
# という雰囲気だったんだけど違うわ

n = int(input())
arr = list(map(int, input().split()))
max_t, max_a = 0, 0
for i in range(n):

    for j in range(n):
        if i == j:
            continue
        tmp_count_t, tmp_count_a = 0, 0
        x, y = i, j
        if i > j:
            x, y = j, i
        for k in range(x, y + 1):
            if k % 2 == 0:
                tmp_count_t += arr[k]
            else:
                tmp_count_a += arr[k]

        if tmp_count_a > max_a:
            max_a = tmp_count_a
            max_t = tmp_count_t
            print("最適解変更: {} 青木p: {}".format(arr[x:y+1], max_a))
    if tmp_count_t > max_t:
        max_t = tmp_count_t

print(max_t)

File: Practice/atcoder/ABC/031/src/c.rb
n = gets.to_i
arr = gets.chomp.split.map(&:to_i)

max_t = -100000
n.times do |i|
  wakaran = []
  n.times do |j|
    next if i == j
    if j < i
      wakaran << arr[j, i-j+1]
    else
      wakaran << arr[i, j-i+1]
    end
  end
  max_a = -1000000
  target = []
  wakaran.each do |w|
    # atcoderはrubyに対して無理解なのか、なぜか2.3を未だに使っているため、
    # Enumerable::sumが使えず、injectで合計を求めている
    temp_max_a = w.select.with_index{|_ ,idx| idx.odd?}.inject(:+)   # zero order
    if max_a < temp_max_a
      max_a = temp_max_a
      target = w
    end
  end
  # atcoderはrubyに対して無理解なのか、なぜか2.3を未だに使っているため、
  # Enumerable::sumが使えず、injectで合計を求めている
  max_t = [max_t, target.select.with_index{|_, idx| idx.even?}.inject(:+)].max # zero order
end
puts max_t
